src/include/futex.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+/*-
+ * Copyright (c) 2024-present MongoDB, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * Futex API is intended for building other synchronization mechanisms, and is not suitable for
+ * general use.
+ */
+
+/*
+ * Linux limits the futex size to 32 bits irrespective of architecture word size.
+ */
+typedef uint32_t WT_FUTEX_WORD;
```

Comment:
When using this type, is there a need to give the compiler a hint that the futex word's address has to be word-aligned?